src/overlays/actors/ovl_En_Wf/z_en_wf.c

```diff
@@ -137,81 +143,1349 @@ static ColliderCylinderInit D_80B37A5C = {
+            this->actor.flags |= 1;
+
+            if ((this->actor.params != 0) && (this->switchFlag != 0xFF)) {
+                func_800F5ACC(0x38); // Mini-Boss Battle Theme
+            }
+        }
+    } else if (this->actionTimer != 0) {
+        this->actor.scale.y += (this->actor.scale.x * 0.2f);
+        this->actor.world.pos.y += 0.5f;
+        Math_SmoothStepToF(&this->actor.shape.shadowScale, 70.0f, 1.0f, 14.0f, 0.0f);
+        this->actionTimer--;
+
+        if (this->actionTimer == 0) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_WOLFOS_APPEAR);
+        }
+    } else if (SkelAnime_Update(&this->skelAnime)) {
+        this->actor.scale.y = this->actor.scale.x;
+        this->actor.gravity = -2.0f;
+        EnWf_SetupWait(this);
+    }
+}
+
+void EnWf_SetupWait(EnWf* this) {
+    Animation_MorphToLoop(&this->skelAnime, &gWolfosWaitingAnim, -4.0f);
+    this->action = WOLFOS_ACTION_WAIT;
+    this->actionTimer = (Rand_ZeroOne() * 10.0f) + 2.0f;
+    this->actor.speedXZ = 0.0f;
+    this->actor.world.rot.y = this->actor.shape.rot.y;
+    EnWf_SetupAction(this, EnWf_Wait);
+}
+
+void EnWf_Wait(EnWf* this, GlobalContext* globalCtx) {
+    Player* player;
+    s32 pad;
+    s16 angle;
+
+    player = GET_PLAYER(globalCtx);
+    SkelAnime_Update(&this->skelAnime);
+
+    if (this->headTilt != 0) {
+        angle = (this->actor.yawTowardsPlayer - this->actor.shape.rot.y) - this->headRot.y;
+
+        if (ABS(angle) > 0x2000) {
+            this->headTilt--;
+            return;
+        }
+
+        this->headTilt = 0;
+    }
+
+    angle = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    angle = ABS(angle);
+
+    if (!(EnWf_DodgeRanged(globalCtx, this))) {
+        // Only use of unk_2E0: never not zero, so this if block never runs
+        if (this->unk_2E0 != 0) {
+            this->unk_2E0--;
+
+            if (angle >= 0x1FFE) {
+                return;
+            }
+            this->unk_2E0 = 0;
+        } else if (EnWf_ChangeAction(globalCtx, this, 0)) {
+            return;
+        }
+
+        angle = player->actor.shape.rot.y - this->actor.shape.rot.y;
+        angle = ABS(angle);
+
+        if ((this->actor.xzDistToPlayer < 80.0f) && (player->swordState != 0) && (angle >= 0x1F40)) {
+            this->actor.shape.rot.y = this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+            EnWf_SetupRunAroundPlayer(this);
+        } else {
+
+            this->actionTimer--;
+
+            if (this->actionTimer == 0) {
+                if (Actor_IsFacingPlayer(&this->actor, 0x1555)) {
+                    if (Rand_ZeroOne() > 0.3f) {
+                        EnWf_SetupRunAtPlayer(this, globalCtx);
+                    } else {
+                        EnWf_SetupRunAroundPlayer(this);
+                    }
+                } else {
+                    EnWf_SetupSearchForPlayer(this);
+                }
+                if ((globalCtx->gameplayFrames & 0x5F) == 0) {
```

Comment:
decimal here prob makes more sense. 
someone is prob gonna ask if mod works, but i dont really care too much at this point myself. we'll go back and fix the ones we can anyway

Reply:
I think the rule seems to be it works if it's `2^n -1`, which this one is not. (Doesn't work here anyway.)